gammapy/spectrum/energy_group.py

```diff
@@ -0,0 +1,534 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+"""
+The main class is ``SpectrumEnergyGroupMaker``
+
+These are helper classes to implement the grouping algorithms,
+they are not part of the public Gammapy API at the moment:
+
+* ``EnergyRange``
+* ``SpectrumEnergyGroup``
+
+"""
+from __future__ import absolute_import, division, print_function, unicode_literals
+from collections import OrderedDict
+import numpy as np
+from astropy.units import Quantity
+from astropy.table import Table
+from ..utils.fits import table_from_row_data
+from ..utils.energy import EnergyBounds
+from ..data import ObservationStats
+from .observation import SpectrumObservation
+
+__all__ = [
+    'SpectrumEnergyGroupMaker',
+    'SpectrumEnergyGroup',
+    'SpectrumEnergyGroups',
+    'calculate_flux_point_binning',
+]
+
+# TODO: improve the code so that this isn't needed!
+INVALID_GROUP_INDEX = -99
+
+
+# TODO: this is used for input at the moment,
+# but for output the `bin_type` field is used.
+# Make up your mind!
+UNDERFLOW_BIN_INDEX = -1
+OVERFLOW_BIN_INDEX = -2
+
+
+class SpectrumEnergyGroupMaker(object):
```

Comment:
I think at the [coding sprint](https://github.com/gammapy/gammapy/wiki/PyGamma-coding-sprint-1) we agreed to not use ``Makers`` in Gammapy.
Why not just ``SpectrumEnergyGrouping``?